mne/viz/evoked.py

```diff
@@ -875,3 +880,164 @@ def plot_snr_estimate(evoked, inv, show=True):
+    .. versionadded:: 0.12.0
+    """
+    import matplotlib.pyplot as plt
+
+    if ts_args is None:
+        ts_args = dict(spatial_colors=True)
+    if topomap_args is None:
+        topomap_args = dict()
+
+    # channel selection
+    # simply create a new evoked object(s) with the desired channel selection
+    evoked = evoked.copy()
+
+    if picks is not None:
+        pick_names = [evoked.info["ch_names"][pick] for pick in picks]
+        evoked.pick_channels(pick_names)
+    if exclude == "bads":
+        exclude = [ch for ch in evoked.info['bads']
+                   if ch in evoked.info['ch_names']]
+    if exclude is not None:
+        evoked.drop_channels(exclude)
+
+    info = evoked.info
+    data_types = ['eeg', 'grad', 'mag', 'seeg']
+    ch_types = set([channel_type(info, idx)
+                    for idx in range(info['nchan'])
+                    if channel_type(info, idx) in data_types])
+    # if multiple sensor types: one plot per channel type, recursive call
+    if len(ch_types) > 1:
+        figs = list()
+        for t in ch_types:  # pick only the corresponding channel type
+            ev_ = evoked.pick_channels([info["ch_names"][idx]
+                                        for idx in range(info['nchan'])
+                                        if channel_type(info, idx) == t],
+                                       copy=True)
+            if len(set([channel_type(ev_.info, idx)
+                        for idx in range(ev_.info['nchan'])
+                        if channel_type(ev_.info, idx) in data_types])) > 1:
+                raise RuntimeError("Possibly infinite loop due to channel "
+                                   "selection problem. This should never "
+                                   "happen! Please check your channel types.")
+            figs.append(_joint_plot(ev_, times=times, title=title, show=show,
+                                    ts_args=ts_args, exclude=list(),
+                                    topomap_args=topomap_args))
+        return figs
+
+    fig = plt.figure()
+
+    # set up time points to show topomaps for
+    times = _process_times(evoked, times, few=True)
+
+    # butterfly/time series plot
+    ts_ax = fig.add_subplot(212)
+    ts_args_pass = dict((k, v) for k, v in ts_args.items()
+                        if k not in ["axes", "show", "colorbar"])
+    ts_args_pass.update(dict(spatial_colors=ts_args.get("spatial_colors",
+                                                        True)))
+    plot_evoked(evoked, axes=ts_ax, show=False, set_tight_layout=False,
+                **ts_args_pass)
+
+    old_title = ts_ax.get_title()
+    ts_ax.set_title('')
+    fig.suptitle(title + "\n" + old_title, y=.95)
+
+    # prepare axes for topomap
+    # slightly convoluted due to colorbar placement and for vertical alignment
+    ts = len(times) + 2
+    map_ax = [plt.subplot(4, ts, x + 2 + ts) for x in range(ts - 2)]
+    cbar_ax = plt.subplot(4, 3 * (ts + 1), 6 * (ts + 1))
```

Comment:
The problem is, I can't set the title in a way so that it's guaranteed to stay at a certain distance to the upper plots. This means if I set it too low, for narrow plots, it's not guaranteed to not overlap with the topoplots.

In `plot_epochs_image*, this is not a problem because we can use the axis title. But that doesn't work here, because the title of the top axes is already in use.

Maybe @jaeilepp has an idea?